### PR TITLE
fix the atomic number 0 issue

### PR DESCRIPTION
**Pull request by felixmusil**
Fix the interoperability with ASE that interpret atom symbol 'X' as atomic number 0 which was not properly handled in spherical invariants (see #288).

At the moment rascal::MaxChemElements is 130 so in principle 12 non physical atomic types are supported. I don't think it is important enough to try to make it more systematic so let's keep this as a future feature.

### CODE REVIEW

**Review by Luthaf — CHANGES_REQUESTED** [review]
The fix looks good to me, I think we should also check somewhere that all species/atomic number values we get are actually below `MaxChemElements`, to prevent silent failure if someone passes a higher value.

**Comment by Luthaf** [comment]
On a more fundamental point of view, I don't understand why we are creating the keys for all possible pairs below `MaxChemElements` (https://github.com/cosmo-epfl/librascal/pull/292/files#diff-9d8839ce162b641d7825d239692b7daaR460-R461) instead of taking the pairs that actually exist in the input? The current code create 8k entries in the map, while we could get away with only ~10 if the input contains 5 species, removing a large number of allocations.

File: src/rascal/representations/calculator_spherical_invariants.hh [hunk 1]
```diff
@@ -426,8 +426,8 @@ namespace rascal {
 
         Key_t sparsified_type{0, 0};
         // there are 118 atomic elements atm
```
Comment (Luthaf) [inline src/rascal/representations/calculator_spherical_invariants.hh:428]: Is this comment outdated? It reads like it is a leftover from when the loop was `for (int sp1{1}; sp1 < 118; sp1++)`

**Comment by felixmusil** [comment]
> On a more fundamental point of view, I don't understand why we are creating the keys for all possible pairs below MaxChemElements

The main reason for this is to keep the logic 'sparcified' vs 'non sparcified' in the initialization of the class: key_map maps to different values depending on this. Also the calculators are thought as unchanging entities after initialization which would break if key_map is modified afterwards. I am not saying your remark should not be followed, just that it warrants some more discussion/thoughts.

> I think we should also check somewhere that all species/atomic number values we get are actually below MaxChemElements, to prevent silent failure if someone passes a higher value

done and tested.

**Review by Luthaf — APPROVED** [review]

**Comment by Luthaf** [comment]
Looks good to me as a fix for the original issue =).

> The main reason for this is to keep the logic 'sparcified' vs 'non sparcified' in the initialization of the class: key_map maps to different values depending on this. Also the calculators are thought as unchanging entities after initialization which would break if key_map is modified afterwards. I am not saying your remark should not be followed, just that it warrants some more discussion/thoughts.

I think this is an unfortunate side effect of having calculator store their data inside the systems instead of managing it separately. Agreed that this should be part of a larger scale discussion

**Review by rosecers — CHANGES_REQUESTED** [review]
Looks good to me! Provided it passes CI I'll approve

**Comment by felixmusil** [comment]
You can approve now then. The CI tests are blocking any merge with master.

**Review by rosecers — APPROVED** [review]

**Pull request merged**